test/system_test.py

```diff
@@ -117,13 +117,15 @@ def test_install_py_gridtools_cpp_install_default():
     spack_install('py-gridtools-cpp')
 
 
-@pytest.mark.parametrize("version", ['1.0.3.7', '1.0.3.9'])
+@pytest.mark.parametrize("version", ['1.0.3.7', '1.0.3.9', '1.0.3.10'])
 def test_install_py_gt4py_for_version(version):
     spack_install(f'py-gt4py @{version}')
 
 
-def test_install_py_icon4py():
-    spack_install('py-icon4py')
+@pytest.mark.parametrize("gt4py_version", ['1.0.3.10'])
+@pytest.mark.parametrize("version", ['main', '0.0.14'])
```

Comment:
Two comments:
We only test fixed version of packages to exclude side-effects of changes to packages outside of spack-c2sm. And version `main` does not qualify for that, since it is a moving target.

Secondly if only `icon4py@0.0.14` and `gt4py@1.0.3.10` are relevant, why not remove the others?